### ISSUE TITLE
Dual license MIT and GPLv3

### ISSUE THREAD

**Issue by yupferris** [logicomacorp/WaveSabre#90]
Doing this would allow us to properly comply with the [VST3 SDK requirements](https://developer.steinberg.help/display/VST/VST+3+Licensing) without signing the proprietary agreement, including [distributing binary plugin builds](#83) which would be licensed as GPLv3 only, while the rest of WS (including the core/player code, which is what users will pack in their binaries) can still be MIT-licensed (or, GPLv3 if they so choose, which is a possibility we've not offered before).

I _think_ this logic is sound, but other opinions/interpretations would be appreciated.

**Comment by kusma** [comment]
I think you can simply MIT license all the non-VST code, as MIT is compatible with GPL. And then you can license the VST as GPLv3.  Not sure actual dual-licensing is needed or even solves anything.

I don't think that hurts any use-cases:
- Intros can be distributed without source, because they don't contain the VST itself.
- Binary distributions of the VSTs would require making the source code available. But this doesn't seem likely to be a problem. Do we even care? The only practical example I can think of would be if someone wants to provide binary versions of a proprietary fork of WaveSabre, which seems... unlikely to me... And if it happens, I'm not sure if it's a good thing ;)

**Comment by yupferris** [comment]
Ah, so you mean simply license the binary builds of the VSTs themselves as GPLv3, but keep all source as MIT-licensed? That would indeed be simpler than dual-licensing (all) code, if that's sufficient to meet the VST3 SDK's GPLv3 license terms, which is the core issue we want to solve here.

**Comment by kusma** [comment]
No, I meant to license the VST specific source code as GPLv3. I don't really mind if you prefer to dual-license it, but I don't think that's necessary.

I don't think there's any need to have MIT license on the VST specific sources, and I don't there's any need to have GPLv3 license on the core sources.

**Comment by yupferris** [comment]
Right, that makes sense.
